modules/persistenceService/src/main/scala/persistenceModel/GithubUser.scala

```diff
@@ -0,0 +1,8 @@
+package persistenceModel
+
+sealed trait GithubUser             extends Product with Serializable
+final case class Id(id: Int)        extends AnyVal
+final case class Name(name: String) extends AnyVal
+
+final case class User(id: Id, name: Name) extends GithubUser
+final case class UserName(name: String)   extends GithubUser
```

Comment:
If I'm not wrong, the value class can't be case classes.

So, would be this more something like:
```suggestion
final class Id(id: Int)        extends AnyVal
final class Name(name: String) extends AnyVal
final class UserName(name: String) extends AnyVal

final case class User(id: Id, name: Name)
```

Reply:
hmm, I think I have seen case classes being used as value classes.
But I am also curios to know why they can't be used as value classes.

Reference - https://github.com/http4s/http4s.g8/blob/0.21/src/main/g8/src/main/scala/%24package__packaged%24/HelloWorld.scala#L16

Reply:
Oh, my bad